### PR TITLE
MFA for App access (WebUI)

### PR DESCRIPTION
**Pull request by Joerger**
Implement [MFA for App access](https://github.com/gravitational/teleport/pull/39536) via the WebUI.

### CODE REVIEW

**Review by gzdunek — APPROVED** [review]
I tested it locally, worked fine.

File: web/packages/teleport/src/services/apps/apps.ts [hunk 1]
```diff
@@ -39,20 +43,37 @@ const service = {
...
-        subjectCookieValue: json.subject_cookie_value as string,
-      }));
+  async createAppSession(params: UrlAppParams) {
+    const resolveApp = {
+      fqdn: params.fqdn,
+      cluster_name: params.clusterId,
+      public_addr: params.publicAddr,
+    };
+
+    // Prompt for MFA is per-session MFA is required for this app.
```
Comment (ravicious) [inline web/packages/teleport/src/services/apps/apps.ts:53]:
```suggestion
    // Prompt for MFA if per-session MFA is required for this app.
```

**Review by ravicious — APPROVED** [review]
Nice, I tested it with [httpbin](https://httpbin.org/) added to my cluster and it works really well.

There are two things that are not necessarily issues, and perhaps even work this way by design, but which could in theory be improved in the future.

1. When I'm not logged in to the Web UI and I go to `app.cluster.com`, I have to tap the key twice. First to sign in and then to create the app session.
2. When I repeatedly click "Launch" from the Web UI for the same app, I have to tap the key each time.

**Comment by Joerger** [comment]
> 1. When I'm not logged in to the Web UI and I go to `app.cluster.com`, I have to tap the key twice. First to sign in and then to create the app session.

This is by design, you'd have to do the same thing if you use a ssh-join-session link.

> 2. When I repeatedly click "Launch" from the Web UI for the same app, I have to tap the key each time.

True, this isn't necessary. Clicking "Launch" will always create a new session even if there is already a valid session cookie in the browser - going straight to `foo-app.example.com` would work without repeated taps. I wonder if it's possible to check the browser cookies first?

I'll leave it out of scope for this PR though.

**Comment by Joerger** [comment]
@r0mant @zmb3 Can I get a flaky test skip?

**Pull request merged**